File: labs/07 Access Control/web-login/php-apache/html/v2/break/break.php
<?php

$DICTIONARY = "words.txt";

// enter the DB password below
$db_password = "0Iw5SIdDykdW0iqH5GIzsA==";

// grab the dictionary words
$words = file($DICTIONARY, FILE_IGNORE_NEW_LINES);

foreach ($words as $password)
{
    echo "Trying: $password on $db_password: ";

    // decrypt the password with the known symmetric key
    $plain_password = openssl_decrypt($db_password, "aes-256-cbc", $password);

    // display the results
    echo "$plain_password\n";
}

?>

File: labs/07 Access Control/web-login/php-apache/html/v2/break/break.py
from Cryptodome.Cipher import AES
from Cryptodome.Util.Padding import unpad
from base64 import b64decode

DICTIONARY = "words.txt";

# enter the DB password below
db_password = "";

# grab the dictionary words
with open(DICTIONARY, "r") as file:
    words = file.read().splitlines()

for password in words:
    print(f"Trying: {password} on {db_password}: ", end="")

    # decrypt the password with the known symmetric key
    cipher = AES.new((bytes(password, "utf-8") + (b"\x00" * (32 - len(password))))[:32], AES.MODE_CBC, b"\x00" * 16)
    try:
        plain_password = unpad(cipher.decrypt(b64decode(db_password)), 16)
    except:
        plain_password = b""

    # display the results
    print(f"{plain_password.decode('utf-8')}")

